test/mpi.py

```diff
@@ -23,36 +23,251 @@
+        # should not throw an error
+        try:
+            mpi_instance = getinstance(mpim.MPI, Local, optionparser)
+            mpi_instance.main()
+        except Exception:
+            self.fail("mympirun raised an exception while running main()")
+
+        optdict = mpi_instance.options.__dict__
+
+        print("args given to mympirunoption: %s, instance options: %s, " % (optionparser.args, optdict))
+
+        for opt in optionparser.args:
+            self.assertFalse(opt in optdict)
+
+    # def test_is_mpirun_for(self):
+    #     """test if _is_mpirun_for returns true when it is given the path of its executable"""
+    #     optionparser = MympirunOption()
+
+    #     exitcode1, _ = run_simple("module purge")
+    #     exitcode2, _ = run_simple("module load cluster/delcatty") # load default cluster
+    #     exitcode3, loadoutput = run_simple("module load " + OpenMPI._mpirun_for[0])
+    #     if exitcode1 + exitcode2 + exitcode3 > 0:
+    #         raise Exception("something went wrong while trying to load OpenMPI module: %s" % loadoutput)
+
+    #     instance = getinstance(OpenMPI, Local, optionparser)
+
+    #     print("mpiscriptname: %s, path: %s, instance mpirun for: %s" %
+    #           (instance._mpiscriptname_for, mpim.which('mpirun'),
+    #            instance._mpirun_for))
+    #     self.assertTrue(instance._is_mpirun_for(mpim.which('mpirun')),
+    #                     msg="mpi instance is not an MPI flavor defined by %s according to _is_mpirun_for, path: %s" %
+    #                     (OpenMPI, mpim.which('mpirun')))
+
+    #     exitcode1, _ = run_simple("module purge")
+    #     exitcode2, _ = run_simple("module load cluster/delcatty") # load default cluster\
+    #     if exitcode1 + exitcode2 > 0:
+    #         raise Exception("something went wrong while trying to reset loaded modules")
+
+    def test_set_omp_threads(self):
+        """test if OMP_NUM_THREAD gets set correctly"""
+        mpi_instance = getinstance(mpim.MPI, Local, MympirunOption())
+        mpi_instance.set_omp_threads()
+        self.assertTrue(getattr(mpi_instance.options, 'ompthreads') is not None, msg="ompthreads was not set")
+        self.assertEqual(os.environ["OMP_NUM_THREADS"], getattr(mpi_instance.options, 'ompthreads', None),
+                         msg="ompthreads has not been set in the environment variable OMP_NUM_THREADS")
+
+    def test_set_netmask(self):
+        """test if netmask matches the layout of an ip adress"""
+        mpi_instance = getinstance(mpim.MPI, Local, MympirunOption())
+        mpi_instance.set_netmask()
+        # matches "IP address / netmask"
+        reg = re.compile(r"\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}/\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}")
+        print("netmask: %s" % mpi_instance.netmask)
+        for substr in string.split(mpi_instance.netmask, sep=":"):
+            self.assertTrue(reg.match(substr), msg="%s does not seem to be a valid address" % substr)
```

Comment:
as we're using it anyway, also check if IPy: https://pypi.python.org/pypi/IPy/
Currently 500.500.500.500 is also valid :wink: